pkg/kubelet/server/stats/volume_stat_calculator.go

```diff
@@ -133,7 +133,16 @@ func (s *volumeStatCalculator) calcAndStoreStats() {
 	var ephemeralStats []stats.VolumeStats
 	var persistentStats []stats.VolumeStats
 	for name, v := range metricVolumes {
-		metric, err := v.GetMetrics()
+		metric, err := func() (*volume.Metrics, error) {
+			startTime := time.Now()
+			defer func() {
+				// if the duration is more than 1 second, record the latest duration it took to get metrics for this volume
+				if time.Since(startTime) > 1*time.Second {
+					klog.V(2).InfoS("Volume stat calculation took too long", "pod", klog.KObj(s.pod), "volumeName", name, "duration", duration)
+				}
+			}()
+			return v.GetMetrics()
```

Comment:
there is already a library for that `utiltrace "k8s.io/utils/trace"` 

one example here
https://github.com/kubernetes/kubernetes/blob/935052185ccb3a9dc5d2f33a893df2c8e9eeb890/pkg/util/iptables/iptables.go#L352-L353

Reply:
Great. Updated